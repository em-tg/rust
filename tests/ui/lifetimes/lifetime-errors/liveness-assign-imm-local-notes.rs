//@ check-pass
// Check notes are placed on an assignment that can actually precede the current assignment
// Don't emit a first assignment for assignment in a loop.

fn test() {
    let x;
    if true {
        x = 1;
    } else {
        x = 2;
        x = 3;      //~ WARNING [E0384]
    }
}

fn test_in_loop() {
    loop {
        let x;
        if true {
            x = 1;
        } else {
            x = 2;
            x = 3;      //~ WARNING [E0384]
        }
    }
}

fn test_using_loop() {
    let x;
    loop {
        if true {
            x = 1;      //~ WARNING [E0384]
        } else {
            x = 2;      //~ WARNING [E0384]
        }
    }
}

fn main() {}
