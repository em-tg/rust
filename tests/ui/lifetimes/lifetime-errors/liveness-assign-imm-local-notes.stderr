warning[E0384]: cannot assign twice to immutable variable `x`
  --> $DIR/liveness-assign-imm-local-notes.rs:11:9
   |
LL |         x = 2;
   |         ----- first assignment to `x`
LL |         x = 3;
   |         ^^^^^ cannot assign twice to immutable variable
   |
   = note: `#[warn(mut_non_mut)]` on by default
help: consider making this binding mutable
   |
LL |     let mut x;
   |         +++

warning[E0384]: cannot assign twice to immutable variable `x`
  --> $DIR/liveness-assign-imm-local-notes.rs:22:13
   |
LL |             x = 2;
   |             ----- first assignment to `x`
LL |             x = 3;
   |             ^^^^^ cannot assign twice to immutable variable
   |
help: consider making this binding mutable
   |
LL |         let mut x;
   |             +++

warning[E0384]: cannot assign twice to immutable variable `x`
  --> $DIR/liveness-assign-imm-local-notes.rs:31:13
   |
LL |             x = 1;
   |             ^^^^^ cannot assign twice to immutable variable
   |
help: consider making this binding mutable
   |
LL |     let mut x;
   |         +++

warning[E0384]: cannot assign twice to immutable variable `x`
  --> $DIR/liveness-assign-imm-local-notes.rs:33:13
   |
LL |             x = 1;
   |             ----- first assignment to `x`
LL |         } else {
LL |             x = 2;
   |             ^^^^^ cannot assign twice to immutable variable
   |
help: consider making this binding mutable
   |
LL |     let mut x;
   |         +++

warning: 4 warnings emitted

For more information about this error, try `rustc --explain E0384`.
